main.go

```diff
@@ -145,3 +185,76 @@ func loadConfig(cfgJSON *extapi.JSON) (customDNSProviderConfig, error) {
 
 	return cfg, nil
 }
+
+func (c *ns1DNSProviderSolver) setNS1Client(ch *v1alpha1.ChallengeRequest, cfg ns1DNSProviderConfig) error {
+	ref := cfg.APIKeySecretRef
+	if ref.Key == "" {
+		return fmt.Errorf(
+			"no key '%s' in secret '%s/%s'",
+			ref.Key,
+			ch.ResourceNamespace,
+			ref.Name,
+		)
+	}
+	if ref.Name == "" {
+		return fmt.Errorf(
+			"no Key '%s' in secret '%s/%s'",
+			ref.Key,
+			ch.ResourceNamespace,
+			ref.Name,
```

Comment:
Is this error message accurate? This seems to be thrown when the name of the secret is missing, not the key.

Reply:
yep, I'll fix it up